layerindex/utils.py

```diff
@@ -335,3 +416,149 @@ def chain_unique(*iterables):
             if k not in seen:
                 seen.add(k)
                 yield item
+
+def setup_core_layer_sys_path(settings, branchname):
+    """
+    Add OE-Core's lib/oe directory to sys.path in order to allow importing
+    OE python modules
+    """
+    core_layer = get_layer(settings.CORE_LAYER_NAME)
+    core_layerbranch = core_layer.get_layerbranch(branchname)
+    if core_layerbranch:
+        core_urldir = core_layer.get_fetch_dir()
+        core_repodir = os.path.join(settings.LAYER_FETCH_DIR, core_urldir)
+        core_layerdir = os.path.join(core_repodir, core_layerbranch.vcs_subdir)
+        sys.path.insert(0, os.path.join(core_layerdir, 'lib'))
+
+
+def run_command_interruptible(cmd):
+    """
+    Run a command with output displayed on the console, but ensure any Ctrl+C is
+    processed only by the child process.
+    """
+    def reenable_sigint():
+        signal.signal(signal.SIGINT, signal.SIG_DFL)
+
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
+    try:
+        process = subprocess.Popen(
+            cmd, cwd=os.path.dirname(sys.argv[0]), shell=True, preexec_fn=reenable_sigint, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+
+        reader = codecs.getreader('utf-8')(process.stdout, errors='surrogateescape')
+        buf = ''
+        while True:
+            out = reader.read(1, 1)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The use of `subprocess.Popen.stdout.read()` might deadlock if any OS pipe buffers fill up and block the child process. We recommend using the `communicate()` API.

[Learn more](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate)